Port xnnpack operators to new registration API

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36800 Port xnnpack operators to new registration API**

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21089650](https://our.internmc.facebook.com/intern/diff/D21089650)

ezyang committed:
Port xnnpack operators to new registration API

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Port xnnpack operators to new registration API"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Port xnnpack operators to new registration API"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21089650](https://our.internmc.facebook.com/intern/diff/D21089650)

[ghstack-poisoned]

ezyang committed:
Update on "Port xnnpack operators to new registration API"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21089650](https://our.internmc.facebook.com/intern/diff/D21089650)

[ghstack-poisoned]

ezyang committed:
Update on "Port xnnpack operators to new registration API"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21089650](https://our.internmc.facebook.com/intern/diff/D21089650)

[ghstack-poisoned]

ezyang committed:
Update on "Port xnnpack operators to new registration API"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21089650](https://our.internmc.facebook.com/intern/diff/D21089650)

[ghstack-poisoned]

ezyang committed:
Update on "Port xnnpack operators to new registration API"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21089650](https://our.internmc.facebook.com/intern/diff/D21089650)

[ghstack-poisoned]

ezyang committed:
Update on "Port xnnpack operators to new registration API"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21089650](https://our.internmc.facebook.com/intern/diff/D21089650)

[ghstack-poisoned]

bhosmer commented:
Here's an error out of `ci/circleci: pytorch_xla_linux_bionic_py3_6_clang9_build` btw - marked upstream but it caught my eye
```
Apr 21 17:59:34 torch_xla/csrc/aten_xla_type_default.cpp:9661:20: error: use of undeclared identifier 'aten'; did you mean 'atan'?
Apr 21 17:59:34 TORCH_LIBRARY_IMPL(aten, XLA, m) {
Apr 21 17:59:34                    ^~~~
Apr 21 17:59:34                    atan
Apr 21 17:59:34 /usr/bin/../lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/math.h:41:12: note: 'atan' declared here
Apr 21 17:59:34 using std::atan;
Apr 21 17:59:34            ^
Apr 21 17:59:34 torch_xla/csrc/aten_xla_type_default.cpp:9661:1: error: C++ requires a type specifier for all declarations
Apr 21 17:59:34 TORCH_LIBRARY_IMPL(aten, XLA, m) {
Apr 21 17:59:34 ^
Apr 21 17:59:34 torch_xla/csrc/aten_xla_type_default.cpp:9661:26: error: use of undeclared identifier 'XLA'
Apr 21 17:59:34 TORCH_LIBRARY_IMPL(aten, XLA, m) {
Apr 21 17:59:34                          ^
Apr 21 17:59:34 torch_xla/csrc/aten_xla_type_default.cpp:9661:31: error: use of undeclared identifier 'm'
Apr 21 17:59:34 TORCH_LIBRARY_IMPL(aten, XLA, m) {
Apr 21 17:59:34                               ^
Apr 21 17:59:34 torch_xla/csrc/aten_xla_type_default.cpp:9661:33: error: expected ';' after top level declarator
Apr 21 17:59:34 TORCH_LIBRARY_IMPL(aten, XLA, m) {
Apr 21 17:59:34                                 ^
Apr 21 17:59:34                                 ;
```

ezyang commented:
Yeah, this is a companion change to xla for the earlier diff in the stack that needs to be landed together

ezyang committed:
Update on "Port xnnpack operators to new registration API"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21089650](https://our.internmc.facebook.com/intern/diff/D21089650)

[ghstack-poisoned]

ezyang committed:
Update on "Port xnnpack operators to new registration API"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21089650](https://our.internmc.facebook.com/intern/diff/D21089650)

[ghstack-poisoned]

ezyang committed:
Update on "Port xnnpack operators to new registration API"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21089650](https://our.internmc.facebook.com/intern/diff/D21089650)

[ghstack-poisoned]
